rosette/src/BaseSupp.cc

```diff
@@ -742,24 +744,29 @@ DEF("regexpCompare", regexpCompare, 2, 2) {
     CHECK(0, RBLstring, re);
     CHECK(1, RBLstring, str);
 
-    char *re_s, *str_s;
-    regexp* r;
     Ob* rv;
 
-    re_s = (char*)&re->byte(0);
-    str_s = (char*)&str->byte(0);
+    char* re_s = (char*)&re->byte(0);
+    char* str_s = (char*)&str->byte(0);
+
+    try {
+        std::regex r(re_s); // Compile the regexp
```

Comment:
If this is the only line where `re_s` is used, then I would write:

```
regex r(re->asCstring());
```

Reply:
Done